src/main/java/com/nike/cerberus/service/DistributedLockService.java

```diff
@@ -136,10 +136,13 @@ public void executeServerShutdownHook(ServerConfig serverConfig, Channel channel
         locks.forEach((name, lock) -> {
             try {
                 lock.release();
-                lock.semaphore.tryAcquire(3, TimeUnit.SECONDS);
+                boolean didAcquireLock = lock.semaphore.tryAcquire(3, TimeUnit.SECONDS);
```

Comment:
Is this the bug SpotBugs found?

Reply:
Yeah, and the bug was real, basically I had assumed the interrupt exception was thrown when the timeout occurred, but in reality it returns false, which was being ignored.